nemo_skills/pipeline/utils/server.py

```
@@ -54,12 +59,53 @@ def get_free_port(exclude: list[int] | None = None, strategy: int | str = 5000)
+            fcntl.flock(lock, fcntl.LOCK_EX)
+            try:
+                with open(state_path) as state:
+                    for line in state:
+                        line = line.strip()
+                        if line.isdigit():
+                            reserved.add(int(line))
+            except FileNotFoundError:
+                pass
+
+            for _ in range(10000):
+                port = random.randint(20000, 65000)
```

Ruff (0.15.15)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)